engine/table/src/main/java/io/deephaven/engine/table/impl/by/ChunkedOperatorAggregationHelper.java

```diff
@@ -608,14 +608,14 @@ private void doRemovesForChunk(@NotNull final RowSequence keyIndicesToRemoveChun
 
         private void propagateRemovesToOperators(@NotNull final RowSequence keyIndicesToRemoveChunk,
                 @NotNull final WritableIntChunk<RowKeys> slotsToRemoveFrom) {
-            final boolean findRuns = ac.requiresRunFinds(SKIP_RUN_FIND);
-            findSlotRuns(sortKernelContext, hashedRunContext, runStarts, runLengths, chunkPositions, slotsToRemoveFrom,
-                    findRuns);
-
+            final boolean permute = findSlotRuns(sortKernelContext, hashedRunContext, runStarts, runLengths,
+                    chunkPositions, slotsToRemoveFrom,
+                    ac.requiresRunFinds(SKIP_RUN_FIND));
 
             if (ac.requiresIndices()) {
-                if (findRuns) {
+                if (permute) {
                     final LongChunk<OrderedRowKeys> keyIndices = keyIndicesToRemoveChunk.asRowKeyChunk();
+                    permutedKeyIndices.setSize(keyIndices.size());
```

Comment:
Was the absence of this `setSize` a bug, or are you just trying to be more hygienic?

Reply:
Permute input does not set the size of the output; so without this the permutedKeyIndices size is undefined.  I don't recall if it manifested itself anywhere.